samples/csharp/end-to-end-apps/ObjectDetection-Onnx/README.md

```diff
@@ -0,0 +1,194 @@
+                .Append(_mlContext.Transforms.ExtractPixels(outputColumnName: "image"))
+                .Append(_mlContext.Transforms.ApplyOnnxModel(modelFile: onnxModelFilePath, outputColumnNames: new[] { TinyYoloModelSettings.ModelOutput }, inputColumnNames: new[] { TinyYoloModelSettings.ModelInput }));
+```
+
+You also need to inspect the neural network to get the names of the input and output nodes. To do this, you can use tools like [Netron](https://github.com/lutzroeder/netron), which is automatically installed with [Visual Studio Tools for AI](https://visualstudio.microsoft.com/downloads/ai-tools-vs/).
+These names are used later when we define the estimation pipeline: in the case of the Tiny Yolo2 network, the input tensor is named **'image'** and the output is named **'grid'**.
+
+We'll use these to define the **input** and **output** parameters of the Tiny Yolo2 Onnx Model.
+
+```csharp
+public struct TinyYoloModelSettings
+{
+    // for checking TIny yolo2 Model input and output parameter names,
+    // you can use tools like Netron, which is installed by Visual Studio AI Tools
+
+    // input tensor name
+    public const string ModelInput = "image";
+
+    // output tensor name
+    public const string ModelOutput = "grid";
+}
+```
+
+![inspecting neural network with netron](./docs/Netron/netron.PNG)
+
+Create the model by fitting the DataView.
+
+```csharp
+var model = pipeline.Fit(dataView);
+```
+
+## Load model and create PredictionEngine
+
+After the model is configured, we need to save the model, load the saved model, create a `PredictionEngine`, and the pass the image to the engine to detect objects using the model.
+This is one place that the **Web** app and the **WPF** app differ slightly.  The **Web** app uses a `PredicitonEnginePool` to efficiently manage and provide the service with a `PredictionEngine` to use to make predictions.
+
+```csharp
+public ObjectDetectionService(PredictionEnginePool<ImageInputData, ImageObjectPrediction> predictionEngine)
+{
+    this.predictionEngine = predictionEngine;
+}
+```
+
+Whereas the **WPF** desktop app creates a single `PredictionEngine` and caches locally to be used for each frame prediction.
+
+```csharp
+public PredictionEngine<ImageInputData, ImageObjectPrediction> GetMlNetPredictionEngine()
+{
+    return _mlContext.Model.CreatePredictionEngine<ImageInputData, ImageObjectPrediction>(_mlModel);
+}
+```
+
+## Detect objects in the image
+
+When obtaining the prediction, we get an `float` array of size **21125** in the `PredictedLabels` property. This is the 125x13x13 output of the model discussed earlier. We then use the `YoloOutputParser` class to interpret and returns a number of bounding boxes for each image. Again these boxes are filtered so that we retrieve only 5 with high confidence.
```

Comment:
```suggestion
When obtaining the prediction, we get an `float` array of size **21125** in the `PredictedLabels` property. This is the 125x13x13 output of the model discussed earlier. We then use the `YoloOutputParser` class to interpret and return a number of bounding boxes for each image. Again, these boxes are filtered so that we retrieve only 5 with high confidence.
```